controllers/control_plane_init.go

```diff
@@ -0,0 +1,48 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/cloudinit"
+)
+
+type UserDataGenerator struct {
+	logr.Logger
+}
+
+func (u *UserDataGenerator) GenerateControlPlaneInit(config *v1alpha1.KubeadmBootstrapConfig) error {
+	u.Info("Generating user data for first control plane")
+
+	// TODO: sub in things in the cluster configuration like the cluster name and other data we can extract from the cluster object
```

Comment:
In this version of the code we don't support custom certificates. However, there is a bunch of defaulting I can do to make this config actually work. The user must still supply

* Controlplane endoint + port formatted correctly
* cloud provider